x-pack/plugins/stack_connectors/server/connector_types/resilient/resilient.ts

```diff
@@ -0,0 +1,363 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AxiosError } from 'axios';
+import { omitBy, isNil } from 'lodash/fp';
+import { CaseConnector, ServiceParams } from '@kbn/actions-plugin/server';
+import { schema } from '@kbn/config-schema';
+import { getErrorMessage } from '@kbn/actions-plugin/server/lib/axios_utils';
+import {
+  CreateIncidentData,
+  ExternalServiceIncidentResponse,
+  GetIncidentTypesResponse,
+  GetSeverityResponse,
+  Incident,
+  ResilientConfig,
+  ResilientSecrets,
+  UpdateIncidentParams,
+} from './types';
+import * as i18n from './translations';
+import { SUB_ACTION } from './constants';
+import {
+  ExecutorSubActionCommonFieldsParamsSchema,
+  ExecutorSubActionGetIncidentParamsSchema,
+  ExecutorSubActionGetIncidentTypesParamsSchema,
+  ExecutorSubActionGetSeverityParamsSchema,
+  ExecutorSubActionHandshakeParamsSchema,
+  GetCommonFieldsResponseSchema,
+  GetIncidentTypesResponseSchema,
+  GetSeverityResponseSchema,
+} from './schema';
+import { formatUpdateRequest } from './utils';
+
+const VIEW_INCIDENT_URL = `#incidents`;
+
+export class ResilientConnector extends CaseConnector<ResilientConfig, ResilientSecrets> {
+  private urls: {
+    incidentTypes: string;
+    incident: string;
+    comment: string;
+    severity: string;
+  };
+
+  constructor(params: ServiceParams<ResilientConfig, ResilientSecrets>) {
+    super(params);
+
+    this.urls = {
+      incidentTypes: `${this.getIncidentFieldsUrl()}/incident_type_ids`,
+      incident: `${this.getOrgUrl()}/incidents`,
+      comment: `${this.getOrgUrl()}/incidents/{inc_id}/comments`,
+      severity: `${this.getIncidentFieldsUrl()}/severity_code`,
+    };
+
+    this.registerSubActions();
+  }
+
+  protected getResponseErrorMessage(error: AxiosError) {
+    if (!error.response?.status) {
+      return i18n.UNKNOWN_API_ERROR;
+    }
+    if (error.response.status === 401) {
+      return i18n.UNAUTHORIZED_API_ERROR;
+    }
+    return `API Error: ${error.response?.statusText}`;
+  }
+
+  private registerSubActions() {
+    this.registerSubAction({
+      name: SUB_ACTION.INCIDENT_TYPES,
+      method: 'getIncidentTypes',
+      schema: ExecutorSubActionGetIncidentTypesParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.SEVERITY,
+      method: 'getSeverity',
+      schema: ExecutorSubActionGetSeverityParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.FIELDS,
+      method: 'getFields',
+      schema: ExecutorSubActionCommonFieldsParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.INCIDENT,
+      method: 'getIncident',
+      schema: ExecutorSubActionGetIncidentParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.HANDSHAKE,
+      method: 'handshake',
+      schema: ExecutorSubActionHandshakeParamsSchema,
+    });
+  }
+
+  private getAuthHeaders() {
```

Comment:
Yes, this was an interesting case. 
So previously, we created `axiosInstance` in service and passed `username` and `password` in auth header of axios:
` const axiosInstance = axios.create({
    auth: { username: apiKeyId, password: apiKeySecret },
  });` 

But now 
- `axiosInstance` is created in `x-pack/plugins/actions/server/sub_action_framework/sub_action_connector.ts` 
- Also if we pass `username` and `password` as request headers to subactionFramework, we will have to add utility to convert it to axios auth header. 

So this seemed a simpler approach to send basic auth as token instead of username or password. This way resilient connector is responsible of its authentication header.

Reply:
I see. The `request` method in `sub_action_connector.ts` accepts all `AxiosRequestConfig`. You could do 

```
this.request({ //my other fields, auth: { username, password } }
```

What you did is working fine and it is better than before. Could you please verify with their documentation that authorizing by headers is ok?

Reply:
Found [this](https://www.ibm.com/docs/en/ibm-mq/9.3?topic=security-using-http-basic-authentication-rest-api) in documentation.

Also tested from our end with Authorization headers and it is working as expected.